Doc/library/typing.rst

```diff
@@ -1266,213 +1323,325 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
-    Using a *constrained* type variable, however, means that the ``TypeVar``
-    can only ever be solved as being exactly one of the constraints given::
+      # Can be anything with an __abs__ method
+      def print_abs[V: SupportsAbs](arg: V) -> None:
+          print("Absolute value:", abs(arg))
 
-       a = concatenate('one', 'two')
-       reveal_type(a)  # revealed type is str
+   .. _typing-constrained-typevar:
 
-       b = concatenate(StringSubclass('one'), StringSubclass('two'))
-       reveal_type(b)  # revealed type is str, despite StringSubclass being passed in
+   Using a *constrained* type variable, however, means that the ``TypeVar``
+   can only ever be solved as being exactly one of the constraints given::
 
-       c = concatenate('one', b'two')  # error: type variable 'A' can be either str or bytes in a function call, but not both
+      a = concatenate('one', 'two')
+      reveal_type(a)  # revealed type is str
 
-    At runtime, ``isinstance(x, T)`` will raise :exc:`TypeError`.  In general,
-    :func:`isinstance` and :func:`issubclass` should not be used with types.
+      b = concatenate(StringSubclass('one'), StringSubclass('two'))
+      reveal_type(b)  # revealed type is str, despite StringSubclass being passed in
 
-    Type variables may be marked covariant or contravariant by passing
-    ``covariant=True`` or ``contravariant=True``.  See :pep:`484` for more
-    details.  By default, type variables are invariant.
+      c = concatenate('one', b'two')  # error: type variable 'A' can be either str or bytes in a function call, but not both
 
-.. class:: TypeVarTuple
+   At runtime, ``isinstance(x, T)`` will raise :exc:`TypeError`.  In general,
+   :func:`isinstance` and :func:`issubclass` should not be used with types.
 
-    Type variable tuple. A specialized form of :class:`type variable <TypeVar>`
-    that enables *variadic* generics.
+   .. attribute:: __name__
 
-    A normal type variable enables parameterization with a single type. A type
-    variable tuple, in contrast, allows parameterization with an
-    *arbitrary* number of types by acting like an *arbitrary* number of type
-    variables wrapped in a tuple. For example::
+      The name of the type variable.
 
-        T = TypeVar('T')
-        Ts = TypeVarTuple('Ts')
+   .. attribute:: __covariant__
 
-        def move_first_element_to_last(tup: tuple[T, *Ts]) -> tuple[*Ts, T]:
-            return (*tup[1:], tup[0])
+      Whether the type variable is covariant.
 
-        # T is bound to int, Ts is bound to ()
-        # Return value is (1,), which has type tuple[int]
-        move_first_element_to_last(tup=(1,))
+   .. attribute:: __contravariant__
 
-        # T is bound to int, Ts is bound to (str,)
-        # Return value is ('spam', 1), which has type tuple[str, int]
-        move_first_element_to_last(tup=(1, 'spam'))
+      Whether the type variable is contravariant.
```

Comment:
Maybe we should say "Whether the type variable has been explicitly marked as covariant/contravariant".

Autovariant TypeVars will still have covariant behaviour if they are inferred by type checkers as being covariant; they just won't be explicitly declared by users to be covariant.

(Maybe we should introduce the term "autovariant typevars" somewhere in this document? It feels kinda hard to talk about TypeVars with inferred variance otherwise. "PEP-695 TypeVars" is clunky, and doesn't cover the fact that old-style TypeVars can have `infer_variance=True` passed in. "TypeVars with inferred variance" is even clunkier.)